staging/src/k8s.io/apiserver/pkg/apis/audit/types.go

```diff
@@ -0,0 +1,200 @@
+
+	Items []Event
+}
+
+// +nonNamespaced=true
+
+// Policy defines the configuration of audit logging, and the rules for how different request
+// categories are logged.
+type Policy struct {
+	metav1.TypeMeta
+
+	// Rules specify the audit Level a request should be recorded at.
+	// A request may match multiple rules, in which case the FIRST matching rule is used.
+	// The default audit level is None, but can be overridden by a catch-all rule at the end of the list.
+	Rules []PolicyRule
+}
+
+// PolicyRule maps requests based off metadata to an audit Level.
+// Requests must match the rules of every field (an intersection of rules).
+type PolicyRule struct {
+	// The Level that requests matching this rule are recorded at.
+	Level Level
+
+	// The users (by authenticated user name) this rule applies to.
+	// An empty list implies every user.
+	// +optional
+	Users []string
+	// The user groups this rule applies to. If a user is considered matching
+	// if the are a member of any of these groups
+	// An empty list implies every user group.
+	// +optional
+	UserGroups []string
+
+	// The verbs that match this rule.
+	// An empty list implies every verb.
+	// +optional
+	Verbs []string
+
+	// Rules can apply to API resources (such as "pods" or "secrets"),
+	// non-resource URL paths (such as "/api"), or neither, but not both.
+	// If neither is specified, the rule is treated as a default for all URLs.
+
+	// Resource kinds that this rule matches. An empty list implies all kinds in all API groups.
+	// +optional
+	ResourceKinds []GroupKinds
+	// Namespaces that this rule matches.
+	// The empty string "" matches non-namespaced resources.
+	// An empty list implies every namespace.
+	// +optional
+	Namespaces []string
+
+	// NonResourceURLs is a set of URL paths that should be audited.
+	// *s are allowed, but only as the full, final step in the path.
+	// Examples:
+	//  "/metrics" - Log requests for apiserver metrics
+	//  "/healthz*" - Log all health checks
+	// +optional
+	NonResourceURLs []string
+}
+
+// GroupKinds represents resource kinds in an API group.
+type GroupKinds struct {
+	// Group is the name of the API group that contains the resources.
+	// The empty string represents the core API group.
+	// +optional
+	Group string
+	// Kinds is a list of kinds of resources within the API group.
+	// Any empty list implies every resource kind in the API group.
+	// +optional
+	Kinds []string
+}
+
+// ObjectReference contains enough information to let you inspect or modify the referred object.
+type ObjectReference struct {
+	// +optional
+	Kind string
+	// +optional
+	Namespace string
+	// +optional
+	Name string
+	// +optional
+	UID types.UID
+	// +optional
+	APIVersion string
+	// +optional
+	ResourceVersion string
+}
+
+// UserInfo holds the information about the user needed to implement the
+// user.Info interface.
+type UserInfo struct {
+	// The name that uniquely identifies this user among all active users.
+	Username string
```

Comment:
This will be filled based on the authorizer attributes, do you know what is used in that case? We do have rbac `ClusterRoleBindings` for controllers, e.g. 

```yaml
apiVersion: rbac.authorization.k8s.io/v1beta1
kind: ClusterRoleBinding
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: system:controller:daemon-set-controller
subjects:
- kind: ServiceAccount
  name: daemon-set-controller
  namespace: kube-system
```

I was thinking the Username in this case would be `daemon-set-controller`. Should we clarify that Username may refer to a service account?

If there actually isn't an authenticated user, we still use the `system:anonymous` user. The one case where this might actually be empty is for requests to the insecure local port (bleh). We should probably define a magic string to use in that case... `system:unauthenticated`?

/cc @liggitt